eclair-core/src/test/scala/fr/acinq/eclair/db/PgUtilsSpec.scala

```diff
@@ -0,0 +1,121 @@
+package fr.acinq.eclair.db
+
+import akka.actor.ActorSystem
+import com.opentable.db.postgres.embedded.EmbeddedPostgres
+import com.typesafe.config.{Config, ConfigFactory}
+import fr.acinq.eclair.TestUtils
+import fr.acinq.eclair.db.pg.PgUtils.JdbcUrlChanged
+import fr.acinq.eclair.db.pg.PgUtils.PgLock.{LockFailure, LockFailureHandler}
+import grizzled.slf4j.Logging
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.io.File
+import java.util.UUID
+
+class PgUtilsSpec extends AnyFunSuite {
+
+  test("database lock") {
+    implicit val system: ActorSystem = ActorSystem()
+    val pg = EmbeddedPostgres.start()
+    val config = PgUtilsSpec.testConfig(pg.getPort)
+    val datadir = new File(TestUtils.BUILD_DIRECTORY, s"pg_test_${UUID.randomUUID()}")
+    datadir.mkdirs()
+    val instanceId1 = UUID.randomUUID()
+    // this will lock the database for this instance id
+    val db = Databases.postgres(config, instanceId1, datadir, LockFailureHandler.logAndThrow)
+
+    assert(
+      intercept[LockFailureHandler.LockException] {
+        // this will fail because the database is already locked for a different instance id
+        Databases.postgres(config, UUID.randomUUID(), datadir, LockFailureHandler.logAndThrow)
+      }.lockFailure === LockFailure.AlreadyLocked(instanceId1))
+
+    // we can renew the lease at will
+    db.obtainExclusiveLock()
+
+    // we wait longer than the lease interval, and make sure that the lock is still there
+    Thread.sleep(20_000)
```

Comment:
Can't we configure a shorter lease interval in tests?

Reply:
We already are using short values (see https://github.com/ACINQ/eclair/blob/e18e81aaba22b24714659a5e7ceb47eae1171a39/eclair-core/src/test/scala/fr/acinq/eclair/db/PgUtilsSpec.scala#L112-L117). We could go a  bit further but I don't want to be too extreme because 5s renew interval is already pretty short. I'l try to reduce them further.

Reply:
Done with 912ba4e.